### PR TITLE
OS support documentation

### PR DESCRIPTION
**Pull request by irozzo-1A**
**What this PR does / why we need it**:
The goal of this PR is to document the OS that machine controller currently supports 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/operating-system.md [hunk 1]
```diff
@@ -4,24 +4,41 @@
...
-| AWS | ✓ | ✓ | ✓ |
-| Openstack | ✓ | ✓ | ✓ |
-| Digitalocean  | ✓ | ✓ | ✓ |
-| Google Cloud Platform | ✓ | ✓ | x |
-| Hetzner | ✓ | x | ✓ |
-| Linode | ✓ | x | x |
+|   | Ubuntu | Container Linux | CentOS | RHEL | SLES |
+|---|---|---|---|---|---|
+| AWS | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Azure | ✓ | ✓ | ✓ | x | x |
```
Comment (moadqassem) [inline docs/operating-system.md:10]: Azure supports rhel os

File: docs/operating-system.md [hunk 2]
```diff
@@ -4,24 +4,41 @@
...
+| Digitalocean  | ✓ | ✓ | ✓ | x | x |
+| Google Cloud Platform | ✓ | ✓ | x | x | x |
```
Comment (moadqassem) [inline docs/operating-system.md:12]: Same here GCE supports rhel

File: docs/operating-system.md [hunk 3]
```diff
@@ -4,24 +4,41 @@
...
+| Hetzner | ✓ | x | ✓ | x | ✓ |
```
Comment (moadqassem) [inline docs/operating-system.md:13]: Hetzner doesn't support SLES

File: docs/operating-system.md [hunk 4]
```diff
@@ -4,24 +4,41 @@
...
+| Packet | ✓ | ✓ | ✓ | x | x |
+| Openstack | ✓ | ✓ | ✓ | ✓ | ✓ |
```
Comment (moadqassem) [inline docs/operating-system.md:15]: Openstack doesn't support SLES

File: docs/operating-system.md (hunk 1 above)
Comment (irozzo-1A) [reply docs/operating-system.md:10]: Ok I'll update

File: docs/operating-system.md (hunk 2 above)
Comment (irozzo-1A) [reply docs/operating-system.md:12]: same here

File: docs/operating-system.md (hunk 4 above)
Comment (irozzo-1A) [reply docs/operating-system.md:15]: well, nothing prevents from using SLES, you mean that it is not covered by tests?

Comment (moadqassem) [reply docs/operating-system.md:15]: It is not covered by the tests and we don't have a prepared image, with cloud-init support added. As the requirement was to run it on one cloud provider only. AWS runs the e2e tests for SLES

Comment (irozzo-1A) [reply docs/operating-system.md:15]: ok

File: docs/operating-system.md (hunk 3 above)
Comment (irozzo-1A) [reply docs/operating-system.md]: ok

**Pull request merged**